src/javascript/javascript-import-scanner.ts

```diff
@@ -254,32 +236,34 @@ export class JavaScriptImportScanner implements JavaScriptScanner {
     if (isWindows() && documentPath.startsWith('/')) {
       documentPath = documentPath.substring(1);
     }
-    let resolvedSpecifier;
-    try {
-      resolvedSpecifier = resolve.sync(specifier, {basedir: documentPath});
-    } catch (e) {
-      warnings.push(new Warning({
-        message: 'Cannot resolve module specifier',
-        sourceRange: document.sourceRangeForNode(node)!,
-        severity: Severity.WARNING,
-        code: 'cant-resolve-module-specifier',
-        parsedDocument: document,
-      }));
-      return undefined;
-    }
 
-    let relativeSpecifierUrl =
-        relative(dirname(documentPath), resolvedSpecifier) as FileRelativeUrl;
-
-    if (isWindows()) {
-      // normalize path separators to URL format
-      relativeSpecifierUrl =
-          relativeSpecifierUrl.replace(/\\/g, '/') as FileRelativeUrl;
+    if (this.moduleResolution === 'node') {
+      try {
+        // TODO (justinfagnani): we need to pass packageName and componentDir,
+        // or maybe the UrlResolver, to get correct relative URLs for
+        // named imports from the root package to its dependencies
+        return resolve(specifier, documentPath);
+      } catch (e) {
+        if (!isPathSpecifier(specifier)) {
+          // Don't warn if the specifier was already a path, even though we do
+          // resolve paths, because maybe the user is serving it some other way.
+          console.warn(`Could not resolve module specifier "${specifier}"`);
```

Comment:
The comment says that we don't warn, but the code does warn. Is this debug code

IMO we should not log to console, we should emit a Warning or throw an Error to express a problem.

Reply:
The comment applies to the if statement (maybe clearer to say "Only warn if the specifier was a name")